lazyflow/utility/io_util/OMEZarrStore.py

```diff
@@ -0,0 +1,181 @@
+    """
+
+    NAME = "OME-Zarr"
+    URL_HINT = 'URL contains "zarr"'
+
+    spec_schema = {
+        "type": "object",
+        "properties": {
+            "multiscales": {
+                "type": "array",
+                "items": {
+                    "type": "object",
+                    "properties": {
+                        "axes": {"type": "array"},
+                        "datasets": {
+                            "type": "array",
+                            "items": {
+                                "type": "object",
+                                "properties": {"path": {"type": "string"}},
+                                "required": ["path"],
+                            },
+                        },
+                    },
+                    "required": ["datasets"],
+                },
+            },
+        },
+        "required": ["multiscales"],
+    }
+
+    def __init__(self, url: str = "", last_scale_only_mode: bool = False):
+        with Timer() as timer:
+            self.url = url
+            uncached_store = FSStore(self.url, mode="r", **OME_ZARR_V_0_4_KWARGS)
+            try:
+                self.ome_spec = json.loads(uncached_store[".zattrs"])
+            except KeyError:
+                raise ValueError("Expected a Zarr store, but could not find .zattrs file at the address.")
+            if _get_ome_spec_version(self.ome_spec) == "0.1":
+                uncached_store = FSStore(self.url, mode="r", **OME_ZARR_V_0_1_KWARGS)
+            self._store = LRUStoreCache(uncached_store, max_size=None)
+            logger.info(f"Initializing OME-Zarr store at {url} took {timer.seconds()*1000} ms.")
+        try:
+            jsonschema.validate(self.ome_spec, self.spec_schema)
+        except jsonschema.ValidationError:
+            raise ValueError(f"Metadata for this dataset did not match OME-Zarr spec.\nReceived:\n{self.ome_spec}.")
+        multiscale_spec = self.ome_spec["multiscales"][0]
+        axistags = _get_axistags_from_spec(multiscale_spec)
+        datasets = multiscale_spec["datasets"]
+        assert len(datasets) > 0, "The OME-Zarr store contains no datasets."
+        dtype = None
+        gui_scale_metadata = {}  # Becomes slot metadata -> must be serializable (no ZarrArray allowed)
+        self._scale_data = {}
+        for scale in reversed(datasets):
+            with Timer() as timer:
+                zarray = ZarrArray(store=self._store, path=scale["path"])
+                dtype = zarray.dtype.type
+                gui_scale_metadata[scale["path"]] = Multiscale(
+                    key=scale["path"], resolution=list(zarray.shape[-1:-4:-1])  # xyz
+                )
+                self._scale_data[scale["path"]] = {
+                    "zarray": zarray,
+                    "chunks": zarray.chunks,
+                    "shape": zarray.shape,
+                }
+                logger.info(f"Initializing scale {scale['path']} took {timer.seconds()*1000} ms.")
```

Comment:
shouldn't seconds -> milliseconds require division by 1000?

```suggestion
                logger.info(f"Initializing scale {scale['path']} took {timer.seconds()/1000} ms.")
```

in case I'm not mistaken here, this would also be applicable to other occurrences in this file

Reply:
awww :/ sorry I was wrong :)

Reply:
I'm also still confused by this 🤷